### PR TITLE
Document KMS integration part 1

### PR DESCRIPTION
**Pull request by osg-grafana**
Partial fix for grafana/grafana-enterprise#2196.

### CODE REVIEW

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+4. From within Grafana, turn on [envelope encryption]({{< relref "../../../administration/envelope-encryption.md" >}}).
+5. Add your AWS KMS details to the Grafana configuration file; depending on your operating system, it is usually named `grafana.ini`:
+ <br><br>a. Add a new section to the configuration file, with a name in the format of `[security.encryption.awskms.<KEY-NAME>]`, where `<KEY-NAME>` is any name that uniquely identifies this key among other provider keys.
+ <br><br>b. Fill in the section with the following values:
+ `key_id`: a reference to a key stored in the KMS. This can be a key ID, a key Amazon Resource Name (ARN), an alias name, or an alias ARN. If you are using an alias, use the prefix “alias/.” To specify a KMS key in a different AWS account, use its ARN or alias. For more information about how to retrieve a key ID from AWS, refer to [Finding the key ID and key ARN](https://docs.aws.amazon.com/kms/latest/developerguide/find-cmk-id-arn.html).
+
+Here are some examples of values you can use in the `key_id` field:
+
+Key ID: `1234abcd-12ab-34cd-56ef-1234567890ab`
+Key ARN: `arn:aws:kms:us-east-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab`
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:34]: For some reason these are rendering without newlines in Github Markdown. Will our docs render them with newlines?

File: docs/sources/enterprise/kms-integration/_index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+# Key management systems (KMSs)
+
+You can choose to encrypt secrets stored in the Grafana database using a key from a KMS, which is a secure central storage location that is designed to help you to create and manage cryptographic keys and control their use across many services. When you integrate with a KMS, Grafana does not directly store your encryption key. Instead, Grafana stores KMS credentials and the identifier of the key, which Grafana uses to encrypt the database.
+
+Grafana integrates with the following key management systems:
+
+* AWS KMS
+* Azure Key Vault
+* [Hashicorp Vault]({{< relref "hashicorp-vault.md" >}})
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/_index.md:16]: I'm conflicted about listing Hashicorp Vault _next to_ AWS KMS and Azure Key Vault, since they are all technically KMS integrations, but Hashicorp Vault is very different from the AWS and Azure integrations. Eventually we will add Hashicorp Vault as a proper KMS integration for database encryption, but until then I think it should remain separate from AWS and Azure. @joanlopez what do you think?

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
++++
+title = "AMS KMS"
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:2]: This is adding AWS KMS content, not refactoring Hashicorp Vault content. Not a huge deal, but should this be a different PR?

Comment (osg-grafana) [reply docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:2]: I changed the name of the PR. :)

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md (hunk 1 above)
Comment (osg-grafana) [reply docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:34]: Yep, need to look at this on Monday.

File: docs/sources/enterprise/kms-integration/_index.md (hunk 2 above)
Comment (undef1nd) [reply docs/sources/enterprise/kms-integration/_index.md:16]: I agree with Mitch that it's a different kind of integration. We have not added Hashicorp Vault as encryption key provider yet.

**Comment by osg-grafana** [comment]
@undef1nd and I will be coordinating some technical changes.

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+2. Retrieve the Key ID.
+ <br><br>In AWS terms, this can be a key ID, a key ARN (Amazon Resource Name), an alias name, or an alias ARN. For more information about how to retrieve a key ID from AWS, refer to [Finding the key ID and key ARN](https://docs.aws.amazon.com/kms/latest/developerguide/find-cmk-id-arn.html).
+
+3. Create a [programmatic credential](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys) (access key ID and secret access key), which has permission to view the key that you created. 
+ <br><br>In AWS, you can control access to your KMS keys by using [key policies](https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html), [IAM policies](https://docs.aws.amazon.com/kms/latest/developerguide/iam-policies.html), and [grants](https://docs.aws.amazon.com/kms/latest/developerguide/grants.html). You can also create [temporary credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html), which must provide a session token along with an access key ID and a secret access key.
```
Comment (undef1nd) [inline docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:23]: I have not tested configuration with temporary credentials. Maybe worth skipping this for now?

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+Update the `[security]` section of the `grafana.ini` configuration file with the new Encryption Provider key that you created:
+
+```
+[security] 
+# previous encryption key, used for legacy alerts, decrypting existing secrets or used as default provider when external providers are not configured
+secret_key = AaaaAaaa
+# encryption provider key in the format <PROVIDER>.<KEY_NAME>
+encryption_provider = awskms.example-encryption-key
```
Comment (undef1nd) [inline docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:65]: `encryption_provider` means current provider used for encryption.
I added one more setting to list all configured providers - `available_encryption_providers`): a user needs to put references to providers sections they configured. 

In your example,`available_encryption_providers ` should be set with the same value as `encryption_provider` - awskms.example-encryption-key. But if they have multiple providers configured, it looks like [here](https://github.com/grafana/grafana-enterprise/pull/2283#issue-782637464):

Comment (undef1nd) [inline docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:65]:
```suggestion
encryption_provider = awskms.example-encryption-key
available_encryption_providers = awskms.example-encryption-key
```

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+available_encryption_providers = awskms.example-encryption-key
```
Comment (undef1nd) [inline docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:66]:
```suggestion
# list of configured key providers, space separated
available_encryption_providers = awskms.example-encryption-key
```

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md (hunk 4 above)
Comment (undef1nd) [reply docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:23]: Filed and issue for this.

File: docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md (hunk 3 above)
Comment (osg-grafana) [reply docs/sources/enterprise/kms-integration/using-aws-kms-to-encrypt-database-secrets.md:2]: Changed the name of the PR again. :)

**Comment by osg-grafana** [comment]
@undef1nd, let's ship this and I will create a new PR based on the work that we have done here.

**Review by undef1nd — APPROVED** [review]
